### DISCUSSION TITLE
Problems getting keymapper to work at all

### DISCUSSION THREAD

**Discussion by Mambouna** [houmain/keymapper#67] [Q&A] [answered]
Hello,

I have problems getting anything to work in keymapper. I suspect the problem is me since I'm a noob at linux (just started trying it out).
I'll include as much information as I can, thank you for your help and patience anyone.
I installed keymapper in the /opt folder, as recommended, since I'm on Fedora.
I placed my keymapper.conf in /etc.
I started the keymapperd service and keymapper from a terminal, with sudo, that was opened in the folder where the files for keymapperd and keymapper are located (because the terminal otherwise didn't recognize the commands).
I know that keymapper and its service are running, because if there is an issue in the config file, I get an error. When there are no issues, there is no error, so I assume that they are running fine.
However, no matter what remappings I try, nothing changes about the output of my inputs.
I have tried the following mappings (I mainly want navigation keys with AltGr+[any keys around J]:
AltRight{J} >> ArrowLeft
AltRight{I} >> ArrowUp
AltRight{L} >> ArrowRight
AltRight{K} >> ArrowDown

J{AltRight} >> ArrowLeft
I{AltRight} >> ArrowUp
L{AltRight} >> ArrowRight
K{AltRight} >> ArrowDown

225{74} >> 37
225{73} >> 38
225{76} >> 39
225{75} >> 40

[modifier="AltRight"]
J >> ArrowLeft
I >> ArrowUp
L >> ArrowRight
K >> ArrowDown

I think the second one is flat out wrong, but I wanted to try everything. The numbers come from the website recommended on this Github page.
Any help and advice appreciated. I googled around but couldn't find something, but If I missed something obvious, just pointing me there would also be really helpful. Thank you.

**Comment by houmain** [comment, marked as answer]
Hi Mambouna, 
at least your first configuration is correct and putting the configuration in /etc/keymapper.conf and starting both keymapperd and keymapper with sudo should work (keymapper should better be startet without sudo).
Have you tried another modifier key? Maybe there is a problem with the AltGr key.
When you start both applications with the -v parameter, more information is output. It should look like this:
```
➜ sudo keymapperd -v        
Waiting for keymapper to connect
Received configuration
Received contexts (1)
Creating uinput device 'Keymapper'
Updating device list
Grabbing device event5 'AT Translated Set 2 keyboard'
Entering update loop
Received contexts (1)
```
```
➜ keymapper -v        
Loading configuration file '/etc/keymapper.conf'
Connecting to keymapperd
Sending configuration
Initializing focused window detection:
  X11 support: initialized
  D-BUS support: initialized
  wlroots support: not available
Entering update loop
```
Cheers

**Reply by Mambouna** [reply]
Hello houmain,
thank you for the reply!
I finally found the issue by appending -v to see where it was loading the configuration file. It seems I had them in both possible directories after trying to get it to work and then only kept editing the one that was in the location that was not being loaded. Now editing the one in the right place, after I corrected some stuff, it works.
Thank you for the quick help, I know it was a dumb mistake but I appreciate the assistance.
Have a good day!
